addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -1,27 +1,425 @@
+        for id_transaction, (invoice_data, attachment) in attachments.items():
+
+            # The server has a maximum number of documents it can send at a time
+            # If that maximum is reached, then we search for more
+            # by re-triggering the download cron, avoiding the timeout.
+            current_num = invoices_data.get('current_num', 0)
+            max_num = invoice_data.get('max_num', 0)
+            retrigger = retrigger or current_num == max_num > 0
+
+            # Import the move from the attachment.
+            # `_create_document_from_attachment` will create an empty move
+            # then try and fill it with the content imported from the attachment.
+            # Should the import fail, thanks to try..except and savepoint,
+            # we will anyway end up with an empty `in_invoice` with the attachment posted on it.
+            # The move will be committed anyway
+            Journal._create_document_from_attachment(attachment.ids)
+            self.env.cr.commit()
+            proxy_acks.append(id_transaction)
+
+        return {"retrigger": retrigger, "proxy_acks": proxy_acks}
+
+    def _l10n_it_edi_get_moves_needing_update(self):
+        self.ensure_one()
+        moves_to_update = self.env['account.move'].search([
+            ('company_id', '=', self.company_id.id),
+            ('l10n_it_edi_transaction', '!=', False),
+            ('l10n_it_edi_state', 'in', WAITING_STATES),
+        ])
+        return {m.l10n_it_edi_transaction: m for m in moves_to_update}
+
+    def _l10n_it_edi_demo_mode_update(self, move=None):
+        self.ensure_one()
+        id_transaction_to_move_map = {move.l10n_it_edi_transaction: move} if move else self._l10n_it_edi_get_moves_needing_update()
+        for id_transaction, move_to_update in id_transaction_to_move_map.items():
+            filename = move_to_update.l10n_it_edi_attachment_id and move_to_update.l10n_it_edi_attachment_id.name or '???'
+            self._l10n_it_edi_process_update(
+                move_to_update,
+                id_transaction,
+                transformed_update_data={
+                    'l10n_it_edi_state': 'forwarded',
+                    'l10n_it_edi_transaction': f'demo_{uuid.uuid4()}',
+                    'send_ack_to_edi_proxy': False,
+                    'date': fields.Date.today(),
+                    'filename': filename},
+                message=_("The e-invoice file %s has been sent in Demo EDI mode.", filename))
+
+    def _l10n_it_edi_update(self, move=None):
+        ''' Check if the sent invoices have been processed by the SdI.
+        '''
+        self.ensure_one()
+        self = self.with_company(self.company_id)
+
+        if self.edi_mode == 'demo':
+            return self._l10n_it_edi_demo_mode_update(move)
+        if move and not move.l10n_it_edi_transaction:
+            # update of a move already rejected
+            return
```

Comment:
Would it make more sense to put this logic in `AccountMove.action_check_l10n_it_edi()`?

Reply:
I've also made a `compute'd field for `l10n_it_edi_waiting_update` so that I can use that in the view also, removing the need for `WAITING_STATES`.